sys-devel/x86_64-multilib-toolchain/x86_64-multilib-toolchain-14.2.0_p1.ebuild

```diff
@@ -0,0 +1,314 @@
+		--enable-kernel=3.2.0
+		--without-selinux
+		--disable-werror
+		--enable-bind-now
+		--enable-fortify-source
+		--disable-profile
+		--without-gd
+		--with-headers="${hdr_dir}"
+		--prefix="${prefix}"
+		--sysconfdir='$(prefix)/etc'
+		--localstatedir='$(prefix)/var'
+		--mandir='$(prefix)/share/man'
+		--infodir='$(prefix)/share/info'
+		--libexecdir='$(libdir)/misc/glibc'
+		--disable-systemtap
+		--disable-nscd
+		--disable-timezone-tools
+	)
+
+	local conf_glibc_amd64=(
+		--libdir='$(prefix)/lib64'
+		--host=${CTARGET}
+	)
+
+	local conf_glibc_x86=(
+		--libdir='$(prefix)/lib'
+		--host=i686-multilib-linux-gnu
+	)
+
+	x86mt-binutils() {
+		# symlink gcc's lto plugin for AR (bug #854516)
+		ln -s ../../libexec/gcc/${CTARGET}/${GCC_PV%%[.-]*}/liblto_plugin.so \
+			"${sysroot}"/lib/bfd-plugins || die
+	}
+
+	# gcc (minimal -- if need more, disable only in stage1 / enable in stage3)
+	local conf_gcc=(
+		--prefix="${prefix}"
+		--host=${CHOST}
+		--disable-bootstrap
+		--disable-cet
+		--disable-gcov #843989
+		--disable-gomp
+		--disable-libquadmath
+		--disable-libsanitizer
+		--disable-libssp
+		--disable-libvtv
+		--disable-werror
+		--with-gcc-major-version-only
+		--with-system-zlib
+		--without-isl
+		--without-zstd
+		--disable-libgomp
+		--enable-poison-system-directories
+	)
+
+	local conf_gcc_stage1=(
+		--disable-cc1
+		--enable-languages=c
+		--disable-libatomic
+		--disable-threads
+		--without-headers
+		--disable-shared
+	)
+
+	local conf_gcc_stage2=(
+		--enable-languages=c,c++
+		--enable-threads
+		--enable-multilib
+		--with-multilib-list=m64,m32
+		--with-build-sysroot="${sysroot}"
+		--with-sysroot="${prefix}"
+		--with-native-system-header-dir="/include"
+	)
+
+	# libstdc++ may misdetect sys/sdt.h on systemtap-enabled system and fail
+	# (not passed in conf_gcc above given it is lost in sub-configure calls)
+	local -x glibcxx_cv_sys_sdt_h=no
+
+	# x86mt-build <path/package-name>
+	# -> ./configure && make && make install && x86mt-package()
+	# passes conf and conf_package to configure, and users can add options
+	# through environment with e.g.
+	#   X86MT_BINUTILS_CONF="--some-option"
+	#   EXTRA_ECONF="--global-option" (generic naming for if not reading this)
+	x86mt-build() {
+		local id=${1##*/}
+		local build_dir=${WORKDIR}/${1}${2+_${2}}-build
+
+		# econf is not allowed in src_compile and its defaults are
+		# mostly unused here, so use configure directly
+		local conf=( "${WORKDIR}/${1}"/configure "${conf[@]}" )
+
+		local -n conf_id=conf_${id} conf_id2=conf_${id}_${2}
+		[[ ${conf_id@a} == *a* ]] && conf+=( "${conf_id[@]}" )
+		[[ ${2} && ${conf_id2@a} == *a* ]] && conf+=( "${conf_id2[@]}" )
+
+		local -n extra_id=X86MT_${id^^}_CONF extra_id2=X86MT_${id^^}_${2^^}_CONF
+		conf+=( ${EXTRA_ECONF} ${extra_id} ${2+${extra_id2}} )
+
+		einfo "Building ${id}${2+ ${2}} in ${build_dir} ..."
+
+		mkdir -p "${build_dir}" || die
+		pushd "${build_dir}" >/dev/null || die
+
+		edo "${conf[@]}"
+		emake MAKEINFO=: V=1
```

Comment:
We need to do something here to account for `/etc/portage/env` and `/etc/portage/make.conf` overriding CC and CXX. This still does not cope at all with either case once we get to building glibc.